File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {

       Queue<TreeNode> queue   = new LinkedList<>();
       List<List<Integer>> res = new LinkedList<>();

       queue.add(root);
       while (!queue.isEmpty()) {

            List<Integer> level = new LinkedList<>();
            int queueSize = queue.size();

            for ( int i = 0; i < queueSize; i++) {
                
                TreeNode node = queue.poll();
                if ( node != null) {
                    queue.add(node.left);
                    queue.add(node.right);
                    level.add(node.val);
                }

            }
            if ( level.size() > 0 ) res.add(level);
       }
       return res;
    }
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:

    def __init__(self):
        self.arr = []

    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.traverse(root)
        return self.arr[k - 1]

    def traverse(self, root: Optional[TreeNode]):
        
        if root is None: return

        self.traverse(root.left)
        self.arr.append(root.val)
        self.traverse(root.right)
